insights/parsers/snmp.py

```diff
@@ -0,0 +1,140 @@
+    ... Ip: 1 64 43767 0 0 0 0 0 41807 18407 12 73 0 0 0 0 0 0 0
+    ... Icmp: InMsgs InErrors InCsumErrors InDestUnreachs InTimeExcds InParmProbs InSrcQuenchs InRedirects InEchos InEchoReps InTimestamps InTimestampReps InAddrMasks InAddrMaskReps OutMsgs OutErrors OutDestUnreachs OutTimeExcds OutParmProbs OutSrcQuenchs OutRedirects OutEchos OutEchoReps OutTimestamps OutTimestampReps OutAddrMasks OutAddrMaskReps
+    ... Icmp: 34 0 0 34 0 0 0 0 0 0 0 0 0 0 44 0 44 0 0 0 0 0 0 0 0 0 0
+    ... IcmpMsg: InType3 OutType3
+    ... IcmpMsg: 34 44
+    ... Tcp: RtoAlgorithm RtoMin RtoMax MaxConn ActiveOpens PassiveOpens AttemptFails EstabResets CurrEstab InSegs OutSegs RetransSegs InErrs OutRsts InCsumErrors
+    ... Tcp: 1 200 120000 -1 444 0 0 6 7 19269 17050 5 4 234 0
+    ... Udp: InDatagrams NoPorts InErrors OutDatagrams RcvbufErrors SndbufErrors InCsumErrors IgnoredMulti
+    ... Udp: 18905 34 0 1348 0 0 0 3565
+    ... UdpLite: InDatagrams NoPorts InErrors OutDatagrams RcvbufErrors SndbufErrors InCsumErrors IgnoredMulti
+    ... UdpLite: 0 0 0 0 0 0 0 0
+    ... '''.strip()
+    >>> from insights.tests import context_wrap
+    >>> shared = {IpSats: IpSats(context_wrap(SNMP_CONTENT))}
+    >>> stats = shared[IpSats]
+    >>> snmp_stats = stats.get_nstats("Ip")
+    >>> print snmp_stats["DefaultTTL"]
+    64
+    >>> snmp_stats = stats.get_nstats("Udp")
+    >>> print snmp_stats["InDatagrams"]
+    18905
+"""
+
+from .. import Parser, parser, LegacyItemAccess
+
+UDPLITE_KEYS = ['InDatagrams', 'NoPorts', 'InErrors', 'OutDatagrams', 'RcvbufErrors', 'SndbufErrors']
+UDP_KEYS = ['InDatagrams', 'NoPorts', 'InErrors', 'OutDatagrams', 'RcvbufErrors', 'SndbufErrors']
+TCP_KEYS = ['RtoAlgorithm', 'RtoMin', 'RtoMax', 'MaxConn', 'ActiveOpens', 'PassiveOpens',
+            'AttemptFails', 'EstabResets', 'CurrEstab    ', 'InSegs', 'OutSegs',
+            'RetransSegs', 'InErrs', 'OutRsts']
+IP_KEYS = ['Forwarding', 'DefaultTTL', 'InReceives', 'InHdrErrors', 'InAddrErrors',
+            'ForwDatagrams', 'InUnknownProtos', 'InDiscards', 'InDelivers', 'OutRequests',
+            'OutDiscards', 'OutNoRoutes', 'ReasmTimeout', 'ReasmReqds', 'ReasmOKs',
+            'ReasmFails', 'FragOKs', 'FragFails', 'FragCreates']
+ICMP_KEYS = ['InMsgs', 'InErrors', 'InDestUnreachs', 'InTimeExcds', 'InParmProbs',
+            'InSrcQuenchs', 'InRedirects', 'InEchos', 'InEchoReps', 'InTimestamps',
+            'InTimestampReps', 'InAddrMasks', 'InAddrMaskReps', 'OutMsgs', 'OutErrors',
+            'OutDestUnreachs', 'OutTimeExcds', 'OutParmProbs', 'OutSrcQuenchs', 'OutRedirects',
+            'OutEchos', 'OutEchoReps', 'OutTimestamps', 'OutTimestampReps', 'OutAddrMasks', 'OutAddrMaskReps']
+
+
+def parse_snmp_contents(content):
+    snmp_stats = {}
+    for line in content:
+        line_split = line.split()
+        if line_split[1].isdigit():
+            if "Ip:" in line_split:
```

Comment:
Have you considered putting these search strings into a list so that you could just iterate over them instead of having all the separate `if` tests?  You could use a tuple to associate the keys with the search string, like `('Ip:', IP_KEYS)`.